### PR TITLE
Fix protocol and domain

### PR DESCRIPTION
**Pull request by jakubgs**
Changes:

* Drop utterly useless `src/common/clients/endpoints/metadata-client-endpoints.js` file
* Use relative API paths in `src/common/clients/metadata-client.js` to be protocol and domain agnostic
* Introduce `back-end/config/index.js` to hold all environment variables used and defaults for them
* Use `node server.js` to run the app instead of `react-scripts start` which is only for devleopment
* Move `react-*` dependencies to `devDependencies` since it's not used once built and deployed

The fact that his was built using [`create-react-app`](https://github.com/facebook/create-react-app) makes modifying this app much harder than it should be. `create-react-app` is intended for learning or fast prototyping, not for building production ready apps.

Fortunately there is `status-react eject`, but it still will pull in a lot of trash. I will probably have to do that in a separate PR.

### CODE REVIEW

File: back-end/config/index.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  DB_CONNECTION                : env.DB_CONNECTION                || null,
+  /* Access */
+  ADMIN_USER                   : env.ADMIN_USER                   || "admin",
+  ADMIN_PASSWORD               : env.ADMIN_PASSWORD               || "discoverbancor",
+  /* IPFS */
+  IPFS_HOST                    : env.IPFS_HOST                    || "ipfs.infura.io",
+  IPFS_PORT                    : env.IPFS_PORT                    || "5001",
+  IPFS_PROTOCOL                : env.IPFS_PROTOCOL                || "https",
+  /* Blockchain */
+  DISCOVER_CONTRACT            : env.DISCOVER_CONTRACT            || "0x25B1bD06fBfC2CbDbFc174e10f1B78b1c91cc77B",
```
Comment (andytudhope) [inline back-end/config/index.js]: Please change this to '0x9fb115BC152aE21ECDc81c73843673fa38e4D2dB`

Comment (jakubgs) [reply back-end/config/index.js]: Changed. Also, use backticks(`) to quote code-like objects, not single quote.

File: back-end/config/index.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  DISCOVER_CONTRACT            : env.DISCOVER_CONTRACT            || "0x9fb115BC152aE21ECDc81c73843673fa38e4D2dB",
+  BLOCKCHAIN_CONNECTION_POINT  : env.BLOCKCHAIN_CONNECTION_POINT  || "wss://ropsten.infura.io/ws/v3/8675214b97b44e96b70d05326c61fd6a",
+  /* EMail */
+  EMAIL_USER                   : env.EMAIL_USER                   || null,
+  EMAIL_PASSWORD               : env.EMAIL_PASSWORD               || null,
+  EMAIL_HOST                   : env.EMAIL_HOST                   || null,
+  EMAIL_PORT                   : env.EMAIL_PORT                   || null,
+  EMAIL_TLS                    : env.EMAIL_TLS                    || null,
+  APPROVER_MAIL                : env.APPROVER_MAIL                || "dapps-approvals@status.im",
+  APPROVE_NOTIFIER_MAIL        : env.APPROVE_NOTIFIER_MAIL        || "dapps-approvals@status.im",
```
Comment (andytudhope) [inline back-end/config/index.js]: Can't we change this to my address, I'm not sure if we ever did set up forwarding, and it seems like I'll be keeping my Status address even though I am a contractor these days?

Comment (jakubgs) [reply back-end/config/index.js]: Not really, I think a group is better. I just checked and you don't appear to be a member:
https://groups.google.com/a/status.im/forum/#!forum/dapps-approvals
I've added you, so now you should be receiving emails sent to that address, as all others in it.
Test email was sent. Next I will check what's up with SMTP setup.

**Pull request merged**